system/lib/wasmfs/fetch_backend.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+// This file defines the JS file backend and JS file of the new file system.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#include "backend.h"
+#include "proxied_async_js_impl_backend.h"
+#include "wasmfs.h"
+
+// See library_wasmfs_fetch.js
+
+extern "C" {
+void _wasmfs_create_fetch_backend_js(wasmfs::backend_t);
+}
+
+namespace wasmfs {
+
+class FetchBackend : public Backend {
+  emscripten::SyncToAsync proxy;
```

Comment:
It seems strange to have a proxying object outside of the virtual proxying backend. Could we use that virtual backend here instead?

Reply:
I guess we can't use the proxying backend here because the proxied operations are not the normal backend operations, but rather async versions of the normal operations. The proxying backend would be expecting to call the normal operations, not async operations.

Reply:
Yes, exactly.

It is odd though to have a Proxy created here, though, so I refactored some now to avoid that. The FetchBackend class is now entirely gone, so creating more backends like this will have less boilerplate.